pandas/io/stata.py

```diff
@@ -2222,17 +2295,46 @@ def _write_bytes(self, value: bytes) -> None:
         """
         self.handles.handle.write(value)  # type: ignore[arg-type]
 
+    def _prepare_non_cat_value_labels(self, data: DataFrame) -> None:
+        """
+        Check for value labels provided for non-categorical columns. Value
+        labels
+        """
+        self._has_value_labels = np.repeat(False, data.shape[1])
+        if self._non_cat_value_labels is None:
+            return
+
+        for labname, labels in self._non_cat_value_labels.items():
+            if labname not in data.columns:
+                # Value label should apply to a column
+                raise ValueError(
+                    f"Can't create value labels for {labname}, it wasn't "
+                    "found in the dataset."
+                )
+            if not is_numeric_dtype(data[labname].dtype):
+                # Labels should not be passed explicitly for categorical
+                # columns that will be converted to int
+                raise ValueError(
+                    f"Can't create value labels for {labname}, value labels "
+                    "can only be applied to numeric columns."
+                )
+            svl = StataNonCatValueLabel(labname, labels)
+            self._value_labels.append(svl)
```

Comment:
Would a dict be simpler here with key labname?

Reply:
Would the dict be for `_value_labels` as { labname: StataValueLabel }? I added the non-categorical value labels onto the existing functions/data used to convert the categoricals columns as much as possible. I'm happy to change this if that makes more sense

Reply:
Probably not worth the effort.